pkg/stores/proxyalpha/proxy_store.go

```diff
@@ -0,0 +1,679 @@
+	AdminK8sInterface() (kubernetes.Interface, error)
+	Client(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+	DynamicClient(ctx *types.APIRequest, warningHandler rest.WarningHandler) (dynamic.Interface, error)
+	AdminClient(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+	TableClient(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+	TableAdminClient(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+	TableClientForWatch(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+	TableAdminClientForWatch(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
+}
+
+type SchemaColumnSetter interface {
+	SetColumns(ctx context.Context, schema *types.APISchema) error
+}
+
+type Cache interface {
+	// ListByOptions returns objects according to the specified list options and partitions
+	// see ListOptionIndexer.ListByOptions
+	ListByOptions(ctx context.Context, lo informer.ListOptions, partitions []partition.Partition, namespace string) (*unstructured.UnstructuredList, string, error)
+}
+
+// WarningBuffer holds warnings that may be returned from the kubernetes api
+type WarningBuffer []types.Warning
+
+// HandleWarningHeader takes the components of a kubernetes warning header and stores them
+func (w *WarningBuffer) HandleWarningHeader(code int, agent string, text string) {
+	*w = append(*w, types.Warning{
+		Code:  code,
+		Agent: agent,
+		Text:  text,
+	})
+}
+
+// RelationshipNotifier is an interface for handling wrangler summary.Relationship events.
+type RelationshipNotifier interface {
+	OnInboundRelationshipChange(ctx context.Context, schema *types.APISchema, namespace string) <-chan *summary.Relationship
+}
+
+type Store struct {
+	clientGetter   ClientGetter
+	notifier       RelationshipNotifier
+	cacheFactory   CacheFactory
+	cfInitializer  CacheFactoryInitializer
+	namespaceCache Cache
+	lock           sync.Mutex
+	columnSetter   SchemaColumnSetter
+}
+
+type CacheFactoryInitializer func() (CacheFactory, error)
+
+type CacheFactory interface {
+	CacheFor(fields [][]string, client dynamic.ResourceInterface, gvk schema.GroupVersionKind, namespaced bool) (factory.Cache, error)
+	Reset() error
+}
+
+// NewProxyStore returns a Store implemented directly on top of kubernetes.
+func NewProxyStore(c SchemaColumnSetter, clientGetter ClientGetter, notifier RelationshipNotifier, factory CacheFactory) (*Store, error) {
+	store := &Store{
+		clientGetter: clientGetter,
+		notifier:     notifier,
+		columnSetter: c,
+	}
+
+	if factory == nil {
```

Comment:
@rmweir IIUC this never happens. Do we need it?

Reply:
[This always happens](https://github.com/rmweir/steve/blob/vai-v2.9/pkg/server/server.go#L252). The ability to pass one mostly helps with testing.